tests/manager/repositories/auth/test_auth_repository.py

```diff
@@ -1,345 +1,494 @@
+
+        # Cleanup
+        async with database_engine.begin_session() as db_sess:
+            await db_sess.execute(
+                association_groups_users.delete().where(
+                    association_groups_users.c.group_id == group_id
+                )
+            )
+            await db_sess.execute(sa.delete(GroupRow).where(GroupRow.id == group_id))
 
     @pytest.mark.asyncio
-    async def test_get_group_membership_validated_not_found(
-        self, auth_repository, mock_db_conn, mocker
-    ):
-        """Test group membership retrieval when not found"""
-        group_id = UUID("87654321-4321-8765-4321-876543218765")
-        user_id = UUID("12345678-1234-5678-1234-567812345678")
+    async def test_get_group_membership_success(
+        self,
+        auth_repository: AuthRepository,
+        sample_user_data: UserTestData,
+        sample_group_data: GroupData,
+    ) -> None:
+        """Test successful group membership retrieval"""
+        result = await auth_repository.get_group_membership(
+            sample_group_data.id, sample_user_data.uuid
+        )
 
-        mocker.patch.object(auth_repository, "_get_group_membership", return_value=None)
+        assert result is not None
+        assert result.group_id == sample_group_data.id
+        assert result.user_id == sample_user_data.uuid
 
-        with pytest.raises(GroupMembershipNotFoundError) as exc_info:
-            await auth_repository.get_group_membership_validated(group_id, user_id)
+    @pytest.mark.asyncio
+    async def test_get_group_membership_not_found(
+        self, auth_repository: AuthRepository, sample_user_data: UserTestData
+    ) -> None:
+        """Test group membership retrieval when not found"""
+        non_existent_group_id = UUID("99999999-9999-9999-9999-999999999999")
 
-        assert "No such project or you are not the member of it" in str(exc_info.value)
+        with pytest.raises(GroupMembershipNotFoundError):
+            await auth_repository.get_group_membership(non_existent_group_id, sample_user_data.uuid)
 
     @pytest.mark.asyncio
-    async def test_check_email_exists_true(self, auth_repository, mock_db_conn):
+    async def test_check_email_exists_true(
+        self, auth_repository: AuthRepository, sample_user_data: UserTestData
+    ) -> None:
         """Test email existence check when email exists"""
-        # Mock query result
-        mock_result = MagicMock()
-        mock_result.first.return_value = MagicMock(email="existing@example.com")
-        mock_db_conn.execute.return_value = mock_result
-
-        result = await auth_repository.check_email_exists("existing@example.com")
+        result = await auth_repository.check_email_exists(sample_user_data.email)
 
         assert result is True
 
     @pytest.mark.asyncio
-    async def test_check_email_exists_false(self, auth_repository, mock_db_conn):
+    async def test_check_email_exists_false(self, auth_repository: AuthRepository) -> None:
```

Comment:
It seems better to have it as "not exists, exists" rather than "exists_true, false." It's unnatural to have a flag value together with true and false.